app/models/miq_task.rb

```diff
@@ -295,6 +296,17 @@ def process_cancel
     end
   end
 
+  def attributes_hash_when_state_change(state, attributes = {})
+    case attributes[:state] = state
+    when STATE_ACTIVE
+      attributes[:started_on] = Time.now.utc if started_on.nil?
```

Comment:
If you go with just settings methods, then you do `self.started_on ||= Time.now.utc`